scheduler/recovery.go

```diff
@@ -2,21 +2,64 @@ package scheduler
 
 import (
 	"context"
+	"errors"
 	"reflect"
+	"time"
+
+	"github.com/puppetlabs/horsehead/v2/netutil"
+)
+
+const (
+	defaultBackoffMultiplier         = time.Millisecond * 5
+	defaultMaxRetries                = 10
+	defaultResetRetriesTimerDuration = time.Second * 10
 )
 
+// ErrMaxRetriesReached is the error returned by RecoveryDescriptor if the max retries
+// have been reached.
+var ErrMaxRetriesReached = errors.New("RecoveryDescriptor: max retries have been reached")
+
+// RecoveryDescriptorOptions contains fields that allow backoff and retry parameters
+// to be set.
+type RecoveryDescriptorOptions struct {
+	// BackoffMultiplier is the timing multiplier between attempts using netutil.Backoff.
+	BackoffMultiplier time.Duration
+	// MaxRetries is the max times the RecoveryDescriptor should attempt to run the delegate
+	// descriptor during a reset retries duration.
+	MaxRetries int
+	// ResetRetriesTimerDuration is the time it takes to reset the retry count when running
+	// a delegate descriptor.
+	ResetRetriesTimerDuration time.Duration
+}
+
 // RecoveryDescriptor wraps a given descriptor so that it restarts if the
 // descriptor itself fails. This is useful for descriptors that work off of
 // external information (APIs, events, etc.).
 type RecoveryDescriptor struct {
-	delegate Descriptor
+	delegate       Descriptor
+	backoff        netutil.Backoff
+	maxRetries     int
+	currentRetries int
```

Comment:
There's actually sort of an unwritten contract (in my mind, not documented, _obviously_ ;) that descriptors should be able to be bound to schedulers multiple times. There might be a better abstraction for this, but in lieu of a formalization of it, we shouldn't use instance variables for anything other than configuration... although luckily I don't think it really needs to be passed around either (see below).